File: Heap(Priority Queue)/506.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        heap = [[-j, i] for i, j in enumerate(score)]
        heapq.heapify(heap)
        cnt = 0
        res = [0] * len(score)
        while heap:
            _, idx = heapq.heappop(heap)                                 
            if cnt == 0:
                res[idx] = "Gold Medal"
            elif cnt == 1:
                res[idx] = "Silver Medal"
            elif cnt == 2:
                res[idx] = "Bronze Medal"
            else:
                res[idx] = str(cnt + 1)
            cnt += 1
        return res

File: Heap(Priority Queue)/506.cpp
class Solution {
public:
    vector<string> findRelativeRanks(vector<int>& score) {
        priority_queue<pair<int,int>> pq;
        vector<string> res(score.size(), "");
        for (int i = 0; i < score.size(); i++) {
            pq.push(make_pair(score[i], i));
        }
        for (int i = 0; i < score.size(); i++) {
            if (i == 0) res[pq.top().second] = "Gold Medal";
            else if (i == 1) res[pq.top().second] = "Silver Medal";
            else if (i == 2) res[pq.top().second] = "Bronze Medal";
            else res[pq.top().second] = to_string(i+1);
            pq.pop();
        }
        return res;
    }
};
